src/view.rs

```diff
@@ -28,6 +29,21 @@ pub fn render_game(app: &mut ApplicationState, ctx: &mut Context) -> GameResult<
     graphics::set_color(ctx, color::SCORE)?;
     graphics::draw(ctx, &text, pt, 0.0)?;
 
+    // Render the gun's heat status in the bottom right of the screen
+    let gun = &app.game_state.world.player.gun;
+    if !gun.is_available() {
+        graphics::set_color(ctx, color::RED)?;
+    }
+
+    let Size { width, height } = app.game_state.world.size;
+    let rect = graphics::Rect {
+        x: width - GUN_HEAT_STATUS_WIDTH - 20.0,
+        y: height - 40.0,
+        w: GUN_HEAT_STATUS_WIDTH * gun.temperature,
+        h: 20.0
+    };
+    graphics::rectangle(ctx, DrawMode::Fill, rect)?;
```

Comment:
I think the heat bar needs a border so the player can understand how far he is from overheat.

It would also be pretty cool if the color changed with heat, maybe go like blue > yellow > orange and red when it overheats.

Reply:
Good idea about the border 👍 
Changing the colour is an easy win too, I'll get onto that

Reply:
Looks really cool!